src/images.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+import { sendRequest } from './util';
+
+export default function images(): Object {
+  const baseUrl = 'images-api.nasa.gov';
+  const endpoints = {
+    search: '/search',
+    asset: '/asset',
+    metadata: '/metadata',
+    captions: '/captions',
+  };
+
+  function validateMediaType(mediaType: string): string {
+    const validMediaTypes = ['image', 'audio'];
+    const i = validMediaTypes.indexOf(mediaType.toLowerCase());
+    return i === -1 ? '' : validMediaTypes[i];
+  }
+
+  function validateYear(year: string): boolean {
+    if (!year || typeof year !== 'string') return false;
+    return year.match(/^\d{4}$/) !== null;
+  }
+
+  return {
+
+    search: (options: Object = {}): Promise<any> =>
+      new Promise((resolve: (data: Object) => void, reject: (reason: Error) => void): mixed => {
+        if (Object.keys(options).length === 0) {
+          return reject(new Error('Expected "q" text search parameter or other keywords'));
```

Comment:
Error message isn't as clear as it could be. Maybe something like "At least one search parameter is required"